### PR TITLE
unsigned int dtypes for numba and numpy interface

### PR DESCRIPTION
**Pull request by t-vi**
- support `__cuda_array_interface__` for numba
- support `torch.tensor(numpy.uint64(42))`
- expand testing of numba and numpy interfaces to include unsigned int dtypes

### CODE REVIEW

**Comment by t-vi** [comment]
@linux-foundation-easycl
/easycla
I think I have signed this now.

**Comment by t-vi** [comment]
The failures are apparently from
- dynamo replacing the inputs with FakeTensor s,
- that gets it a match in the signature, 
- but then the overflow is undetected (and it seems to get into a place where it assumes the replacing of fake with real has happened / there are only the possibilities to see number, list or tuple and actually sees a faketensor).

File: torch/csrc/utils/python_numbers.h [hunk 1]
```diff
@@ -93,7 +93,17 @@ inline uint32_t THPUtils_unpackUInt32(PyObject* obj) {
...
+#ifdef USE_NUMPY
+  // PyLong_AsUnsignedLongLong does not deal with numpy uint64 scalar (e.g.
+  // numpy.uint64(1))
+  if (torch::utils::is_numpy_int(obj) &&
+      (torch::utils::numpy_scalar_scalar_type(obj) == at::kUInt64)) {
+    torch::utils::numpy_scalar_to_c(obj, &value);
+    return value;
+  }
+#endif
+  value = PyLong_AsUnsignedLongLong(obj);
```
Comment (guilhermeleobas) [inline torch/csrc/utils/python_numbers.h:106]: I guess the error is due to not checking if an overflow happened when using `numpy_scalar_to_c`. `PyArray_ScalarAsCtype` does not check for an overflow, but `PyLong_AsUnsignedLongLong` does.

**Review by guilhermeleobas — COMMENTED** [review]
The numba/numpy changes seems ok to me.

File: torch/csrc/utils/python_numbers.h (hunk 1 above)
Comment (lezcano) [reply torch/csrc/utils/python_numbers.h:106]: @t-vi if you address this comment we should be able to land this PR

**Pull request closed without merging**